filament/src/MaterialInstance.cpp

```diff
@@ -183,6 +183,34 @@ template UTILS_PUBLIC void MaterialInstance::setParameter<mat4f>   (const char*
 
 // ------------------------------------------------------------------------------------------------
 
+template<typename T>
+T FMaterialInstance::getParameterImpl(std::string_view name) const noexcept {
+    ssize_t offset = mMaterial->getUniformInterfaceBlock().getFieldOffset(name, 0);
+    return downcast(this)->getUniformBuffer().getUniform<T>(offset);
```

Comment:
I think we might want to add a check here if the offset value is correct? 
What error would we throw in case of an invalid offset?

Reply:
`getFieldInfo()` already validates that `name` exists and throws an exception if it doesn't. So you need to make this function not `noexcept`. I think you can just add an `assert_invariant(offset>=0)` because I don't think it's possible to get there with a bad offset.

Reply:
added!